Makefile

```diff
@@ -73,6 +76,15 @@ micropython: build-$(BOARD_SAFE) wasp/boards/$(BOARD_SAFE)/watch.py micropython/
 		--application micropython/ports/nrf/build-$(BOARD)-s132/firmware.hex \
 		build-$(BOARD)/micropython.zip
 
+wasp/boards/manifest_user_apps.py: wasp.toml
+	pip3 install tomli
```

Comment:
I don't think its appropriate to install packages from a build rule.

Reply:
I 100% agree but there is a bug between pip, docker and the tomli package that results in tomli not being installed correctly within the docker image build. This bug is not present on local or workflows builds. As a result I had to put a pip install command in the make file to ensure the package was installed at compile time.

I worked for hours to find another way around this bug.

- I tried the toml package (the other toml lib) and had the same issue. I think one is based on the other. 
- I tried putting the install on a new layer of the docker image
- Tried different orders of the image
- Tried all the docker ubuntu images to try and find one without the bug
- Tried debian images instead of ubuntu

It's possible its related to https://github.com/pypa/setuptools_scm/issues/625 but that bug was fixed a while ago so im not sure.

One solution is to require python 3.11 (which has tomli built in) but that version is pretty bleeding edge for local dev and it would require tomli be imported as tomlib so it puts us in a weird position of either requiring python 3.11 or doing some kind of conditional import. 

I felt this was the least evil solution. Maybe i should put a comment in the makefile tho?

Reply:
I would be interested if you experience this bug when building via docker with is line removed. Or if you have any ideas on where else this command could be added. To be honest im at a bit of a loss as to how to fix this. Having that install command in the makefile is not great.

I run Docker version 20.10.17, build 100c701 on a Debian based system

Reply:
I have to admit I never use or test docker build code (because it isn't compatible with podman)... so if you had been less conscientious I would never have noticed and it would probably have been left to the docker users to figure this out!

However... I think you can probably work around this by upgrading the Dockerfile to `FROM ubuntu:jammy` and add python3-tomli to the package list (instead of using pip3).

Reply:
I figured out what was happening. I was running the shell script with sudo which resulted in the docker build command failing but then the docker run command was just using an old image. So i never got my updates. 

Removed that line from the makefile. it wasn't a real bug, just me being dumb lol.